modules/services/imapnotify.nix

```diff
@@ -0,0 +1,94 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.imapnotify;
+
+  imapnotifyAccounts =
+    filter (a: a.imapnotify.enable) (attrValues config.accounts.email.accounts);
+
+  genAccountUnit = account: {
+    name = "imapnotify@${account.address}";
+    value = {
+      Unit = {
+        Description = "Execute scripts on IMAP mailbox changes (new/deleted/updated messages) using IDLE for %i";
+        PartOf = [ "network-online.target" ];
```

Comment:
You could add
```ini
After = [ "gpg-agent.service" ];
```
here and the service should scheduled after gpg-agent if both are started.